### ISSUE TITLE
Network setting on display doesn't work.

### ISSUE THREAD

**Issue by riegaz** [billkenney/update_max3_plus3#5]
I tried to join a wifi via the display and it didn't work as it just shows "IP:"

I joined it using "sudo nmtui" but it would be nice if we could get it working.

I was wondering if this has to do with it?

![CleanShot 2024-06-06 at 12 38 59@2x](https://github.com/billkenney/update_max3_plus3/assets/1749401/95e03dea-bdbe-4989-b91a-877b49b82b48)

**Comment by billkenney** [comment]
I'm not sure why the WiFi doesn't work on the display. I added the nmtui section because I thought this may be an issue, although I don't use WiFi. Were you trying to setup WiFi with the printer also plugged into Ethernet? If so, I'd suggest unplugging Ethernet and trying again. I think I've seen some people post that it does not work with Ethernet plugged in, even with the stock software. If that doesn't work, I guess you could try installing the Qidi patch files?

That's strange that ifconfig doesn't work for you, it works on my printer (see image). I'm assuming you wrote the image to your emmc and did not follow the manual guide? If so, the same applications are installed. Maybe try `sudo apt install net-tools` and see if it installs anything

It actually looks like zsh (the shell that I use) may not have loaded for you. When I ssh into my printer it doesn't show mks@mkspi:~, it shows -> ~

Try running `exec zsh` then run `ifconfig` and see if that works. If it doesn't, you can `nano ~/.zshrc` and add this to the bottom of the file: `export PATH=$PATH:/sbin:/usr/sbin`, save, and run `exec zsh` again and it should work

What printer do you have btw? ![image](https://github.com/billkenney/update_max3_plus3/assets/30010560/f5967370-7815-4149-89cd-3b24a83ae48e)

**Comment by riegaz** [comment]
It is the Qidi Max3. I followed every step in your guide and there have been no errors except for the rpi mcu which worked the second time.

ifconfig works when I call it using the entire path /usr/sbin/ifconfig. But it seems that /usr/sbin is not in path so I wondered if that would cause the display issue?

Does the display show the wifi networks for you?

**Comment by billkenney** [comment]
@riegaz did you try running `exec zsh` before `ifconfig`? Or adding the PATH setting to ~/.zshrc?

I'm waiting on a part for my hotend, so my printer is down at the moment. I can check the WiFi on the screen in a few days.

**Comment by riegaz** [comment]
No, I didn't try zsh. I can call ifconfig by running /usr/sbin/ifconfig without any issue.

Not sure if it would help if I would add it to the path as the issue can be everywhere. It is even throwing this at me: 
![CleanShot 2024-06-06 at 19 52 04@2x](https://github.com/billkenney/update_max3_plus3/assets/1749401/d48d81ed-0e52-4875-81a2-e200911f6372)

**Comment by riegaz** [comment]
Is there a way to see what the display executes on the armbian to get this information? I don't see any error messages in dmesg. That would help to solve the issue.

**Comment by billkenney** [comment]
I get an IRQ error every time I start my printer too, but it only happens once and it's shortly after turning the printer on. Someone in another issue also posted about the IRQ error, so it seems like it happens with every one of these printers. I was never able to trace the source, so I just ignore it. It happened before updating to Bookworm, so it has nothing to do with the update process. The only difference I noticed is that the number changed after updating to Bookworm.

**Comment by riegaz** [comment]
Overall, I'm happy that otherwise everything seems to work out. Would be great if preview and network settings would work at some point but both is not urgent. Thanks for providing the guide!

**Comment by billkenney** [comment]
The network menu doesn't work for me either. It did show 192.168.1.2 a few times when I touched the icon, but it wasn't connected. I'll add this as a note to the repo. 

https://github.com/billkenney/update_max3_plus3/assets/30010560/d5ffc985-c0e6-479f-b488-e8d8c9e32b46

**Comment by billkenney** [comment]
@riegaz Qidi (not officially, but one of its employees) released a bookworm image for the max3. Based on the description, it seems like it fixes the image preview and WiFi issues. 

https://github.com/CChen616/QIDI_Max3_Bookworm
